flink-runtime/src/main/java/org/apache/flink/runtime/asyncprocessing/functions/DeclaringAsyncKeyedCoProcessFunction.java

```diff
@@ -0,0 +1,170 @@
+     *     invocation of this method, do not store it.
+     * @param out The collector for returning result values.
+     * @throws Exception This method may throw exceptions. Throwing an exception will cause the
+     *     operation to fail and may trigger recovery.
+     */
+    public void onTimer(long timestamp, OnTimerContext ctx, Collector<OUT> out) throws Exception {}
+
+    /**
+     * Information available in an invocation of {@link #processElement1(Object, Context,
+     * Collector)}/ {@link #processElement2(Object, Context, Collector)} or {@link #onTimer(long,
+     * OnTimerContext, Collector)}.
+     */
+
+    /**
+     * Declaring variables before {@link #declareProcess1} and {@link #declareProcess2} and {@link
+     * #declareOnTimer}.
+     */
+    public void declareVariables(DeclarationContext context) {}
+
+    /**
+     * Declare a process for one element from the first of the connected streams.
+     *
+     * <p>This function can output zero or more elements using the {@link Collector} parameter and
+     * also update internal state or set timers using the {@link Context} parameter.
+     *
+     * @param context the context that provides useful methods to define named callbacks.
+     * @param ctx A {@link Context} that allows querying the timestamp of the element and getting a
+     *     {@link TimerService} for registering timers and querying the time. The context is only
+     *     valid during the invocation of this method, do not store it.
+     * @param out The collector for returning result values.
+     * @return the whole processing logic just like {@code processElement}.
+     */
+    public abstract ThrowingConsumer<IN1, Exception> declareProcess1(
+            DeclarationContext context, Context ctx, Collector<OUT> out)
+            throws DeclarationException;
+
+    /**
+     * Declare a process for one element from the second of the connected streams.
+     *
+     * <p>This function can output zero or more elements using the {@link Collector} parameter and
+     * also update internal state or set timers using the {@link Context} parameter.
+     *
+     * @param context the context that provides useful methods to define named callbacks.
+     * @param ctx A {@link Context} that allows querying the timestamp of the element and getting a
+     *     {@link TimerService} for registering timers and querying the time. The context is only
+     *     valid during the invocation of this method, do not store it.
+     * @param out The collector for returning result values.
+     * @return the whole processing logic just like {@code processElement}.
+     */
+    public abstract ThrowingConsumer<IN2, Exception> declareProcess2(
+            DeclarationContext context, Context ctx, Collector<OUT> out)
+            throws DeclarationException;
+
+    /**
+     * Declare a procedure which is called when a timer set using {@link TimerService} fires.
```

Comment:
the words say it is declaring a procedure but the output parameter says it is  a processor. I would expect these words to be the same